### PR TITLE
Add a timestamp to publication acks

### PR DESCRIPTION
**Pull request by Jmgr**
Publication ACKs currently return the stream offset the message was committed to, but not its timestamp. This PR adds this.

### CODE REVIEW

File: api.proto [hunk 1]
```diff
@@ -192,6 +192,7 @@ message Ack {
     string    ackInbox         = 5; // NATS subject to publish acks to
     string    correlationId    = 6; // User-supplied value from the message
     AckPolicy ackPolicy        = 7; // The AckPolicy sent on the message
+    int64     timestamp        = 8 [jstype=JS_STRING]; // When the message was committed
```
Comment (tylertreat) [inline api.proto]: Based on the implementation in https://github.com/liftbridge-io/liftbridge/pull/254, the timestamp being returned is not when the message was committed but rather when it was _received by the server_.

**Comment by tylertreat** [comment]
Looks like this has conflicts now as well.

**Comment by Jmgr** [comment]
> Looks like this has conflicts now as well.

Should be fixed now.

**Comment by tylertreat** [comment]
@Jmgr Did you see my comment above regarding the `timestamp` semantics? Is your intention to expose a _commit_ time specifically? If so, I don't think the current implementation does this.

**Comment by Jmgr** [comment]
Yes, I was looking at ways to fix that (expose the commit time). Could setting the timestamp when calling `sendAck` work?

**Comment by tylertreat** [comment]
> Could setting the timestamp when calling sendAck work?

Yeah, this would work and is simple. It might be interesting to expose both _received_ time and _commit_ time which would give insight into any lag in committing. I imagine this could be useful for monitoring purposes.

**Pull request merged**